app/electron/main.ts

```diff
@@ -665,6 +665,80 @@ function startElecron() {
       options: {};
     }
 
+    /**
+     * Ask the user with an electron dialog if they want to allow the command
+     * to be executed.
+     * @param command - The command to show in the dialog.
+     * @returns true if the user allows the command to be executed, false otherwise.
+     */
+    function confirmCommandDialog(command: string): boolean {
+      if (mainWindow === null) {
+        return false;
+      }
+      const resp = dialog.showMessageBoxSync(mainWindow, {
+        title: i18n.t('Consent to command being run'),
+        message: i18n.t('Allow this local command to be executed? Your choice will be saved.'),
+        detail: command,
+        type: 'question',
+        buttons: [i18n.t('Allow'), i18n.t('Deny')],
+      });
+
+      return resp === 0;
+    }
+
+    const SETTINGS_PATH = path.join(app.getPath('userData'), 'settings.json');
+
+    /**
+     * Loads the user settings.
+     * If the settings file does not exist, an empty object is returned.
+     * @returns The settings object.
+     */
+    function loadSettings() {
+      try {
+        const data = fs.readFileSync(SETTINGS_PATH, 'utf-8');
+        return JSON.parse(data);
+      } catch (error) {
+        return {};
+      }
+    }
+
+    /**
+     * Saves the user settings.
+     * @param settings - The settings object to save.
+     */
+    function saveSettings(settings) {
```

Comment:
Indeed. This will come in a future PR. There's some description of how that will work in the issue #1744 

In short, there will be a place in the General Settings where folks can see these decisions and change them.